git/lib/git_update.rb

```diff
@@ -26,6 +26,12 @@ def initialize(repo_path, key_id, refs)
   end
 
   def exec
+    Semaphore.exclusively("action:#{@repo_name}") do
```

Comment:
Please find a way to only write `"action:"` once. Maybe put it into a constant or a class method of `GitRepository`.
And please use `git` instead of `action` (we will have many locks with different types and all of them are some kind of action)